### ISSUE TITLE
VPF/NEON detection issue and bus error on Vero4k.

### ISSUE THREAD

**Issue by hissingshark** [hrydgard/ppsspp#15499]
### Game or games this happens in

All.

### What area of the game / PPSSPP

I had been limping on with an old build from 179cafc (?July 2019).  The GTA and Burnout games were glitched.  As soon as you started a map the car/character would fall impossibly through the ground and then it would crash!  Wasn't sure what the regression was and suspected updated GPU drivers.

I finally got around to a rebuild from master, to see if things had sorted themselves out.
The PPSSPP frontend and menus work fine.

---
When loading a ROM it asserts:
>VMOV_neon requires VFP or NEON

Looking at the ArmCPUDetect this is understandable as my `/proc/cpuinfo` only reports:
> Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32

I'm on the S905X, so an Arm Cortex-A53.  Presumably this is misreporting by the kernel or something to do with running 32bit userland?  But it's not new, and I can't see any commits to change the checks since my old build - so not sure why this is only a problem now?

I couldn't work it out so temporarily bypassed with:
```
bVFP = true;
bNEON = true;
```
---
Games now start to load, but by the time they reach a title page they crash with a **bus error**.  The only workaround is to load from a savestate.  In most cases it plays on after that, but some will eventually succumb to the bus error.

Is this anything at all to go on, or do you need my to start builds to bisect it?

Many thanks in advance,

### What should happen

Should accept CPU capabilities and not crash with a bus error (would be nice if GTA and Burnout games weren't knackered either!)

### Platform

Linux / BSD

### Mobile phone model or graphics card

Vero4K (S905x with Cortex A53 and Mali-450 MP3 )

### PPSSPP version affected

e0e3223

### Last working version

179cafc

### Graphics backend (3D API)

OpenGL / GLES

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

**Comment by unknownbrackets** [comment]
Recently, support for non-NEON ARM was dropped.

There was a bug in the commit you tried, which might be causing bus errors.  Could you try the latest?

I wonder if it's worth just checking for "Vortex A53" and forcing NEON on...

-[Unknown]

**Comment by hissingshark** [comment]
Ha, just my luck!  You were right, the bus error had been fixed in the meantime.  Sorry about that.
I'm glad to report that the Burnout/GTA bugs are also fixed.  Good work.

I noticed the dropped non-NEON support and didn't think that applied to me.  But actually the only reason I've been getting by all these years with my misreported spec is because it was allowed...

I've submitted a PR for the detection (#15505) if that's any good to you.

Many thanks for the support.
